cmd/osbuild-worker/jobimpl-osbuild_test.go

```diff
@@ -0,0 +1,81 @@
+package main_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/osbuild/images/pkg/osbuild"
+
+	main "github.com/osbuild/osbuild-composer/cmd/osbuild-worker"
+)
+
+func TestMakeJobErrorFromOsbuildOutput(t *testing.T) {
+	tests := []struct {
+		inputData *osbuild.Result
+		expected  string
+	}{
+		{
+			inputData: &osbuild.Result{
+				Success: false,
+				Log: map[string]osbuild.PipelineResult{
+					"fake-os": []osbuild.StageResult{
+						{
+							Type:    "good-stage",
+							Success: true,
+							Output:  "good-output",
+						},
+						{
+							Type:    "bad-stage",
+							Success: false,
+							Output:  "bad-failure",
+						},
+					},
+				},
+			},
+			expected: `Code: 10, Reason: osbuild build failed in stage:
+bad-stage, Details: []`,
+		},
+		{
+			inputData: &osbuild.Result{
+				Success: false,
+				Log: map[string]osbuild.PipelineResult{
+					"fake-os": []osbuild.StageResult{},
+				},
+			},
+			expected: `Code: 10, Reason: osbuild build failed, Details: []`,
```

Comment:
I hope errors like this do not happen for real, i.e. failure with no further details :) but testing it here makes sense of course.